src/components/HomePage/PlayRadioButton.tsx

```diff
@@ -1,55 +1,64 @@
 
 import PauseIcon from '../../../public/icons/pause.svg';
 import PlayIcon from '../../../public/icons/play-arrow.svg';
-import { triggerPauseAudio } from '../AudioPlayer/EventTriggers';
 import Button from '../dls/Button/Button';
-import Link from '../dls/Link/Link';
-import useCurrentStationInfo from '../Radio/useStationInfo';
+import Spinner from '../dls/Spinner/Spinner';
+import { getRandomCuratedStationId } from '../Radio/curatedStations';
+import { StationType } from '../Radio/types';
 
 import styles from './PlayRadioButton.module.scss';
+import RadioInformation from './RadioInformation';
 
-import { playFrom, selectIsPlaying, selectIsRadioMode } from 'src/redux/slices/AudioPlayer/state';
-import { selectRadioStation } from 'src/redux/slices/radio';
 import { logEvent } from 'src/utils/eventLogger';
+import { selectIsLoading } from 'src/xstate/actors/audioPlayer/selectors';
+import { AudioPlayerMachineContext } from 'src/xstate/AudioPlayerMachineContext';
 
 const PlayRadioButton = () => {
   const { t } = useTranslation('radio');
-  const dispatch = useDispatch();
-  const isAudioPlaying = useSelector(selectIsPlaying);
-  const isRadioMode = useSelector(selectIsRadioMode);
-  const stationState = useSelector(selectRadioStation, shallowEqual);
-  const stationInfo = useCurrentStationInfo();
+  const audioService = useContext(AudioPlayerMachineContext);
 
-  const shouldShowStationName = isRadioMode && isAudioPlaying;
+  const isAudioPlaying = useSelector(audioService, (state) =>
+    state.matches('VISIBLE.AUDIO_PLAYER_INITIATED.PLAYING'),
+  );
+  const isRadioMode = useSelector(audioService, (state) => !!state.context.radioActor);
+  const isLoading = useSelector(audioService, selectIsLoading);
 
+  // TODO: handle continue radio from last saved session
   const onPlayClicked = () => {
+    if (isRadioMode) {
+      audioService.send('TOGGLE');
+      return;
+    }
+    const randomStationId = getRandomCuratedStationId();
+
     logEvent('play_radio_clicked', {
-      stationId: stationState.id,
-      type: stationState.type,
+      stationId: randomStationId,
+      type: StationType.Curated,
     });
 
-    dispatch(
-      playFrom({
-        chapterId: Number(stationState.chapterId),
-        reciterId: Number(stationState.reciterId),
-        shouldStartFromRandomTimestamp: true,
-        isRadioMode: true,
-      }),
-    );
+    audioService.send({
+      type: 'PLAY_RADIO',
+      stationId: Number(randomStationId),
+      stationType: StationType.Curated,
+    });
   };
 
   const onPauseClicked = () => {
-    triggerPauseAudio();
+    audioService.send('TOGGLE');
   };
 
+  const { radioActor } = audioService.getSnapshot().context;
   return (
     <div className={styles.container}>
       <div className={styles.playRadioSection}>
         {isAudioPlaying && isRadioMode ? (
           <Button
-            prefix={<PauseIcon />}
+            prefix={isLoading ? <Spinner /> : <PauseIcon />}
```

Comment:
Since we are already inside `isAudioPlaying`, `isLoading` will never be `true`.

Reply:
maybe it still can? It can go to PLAYING.LOADING and PLAYING.ACTIVE?